application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,265 @@
+     * @param description for the embedded message. nullable {@link String}
+     * @return the formatted embed {@link MessageEmbed}
+     */
+    private static @NotNull MessageEmbed makeEmbed(@Nullable String title,
+            @Nullable String description) {
+
+        if (title == null) {
+            title = "Select your roles:";
+        }
+
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle(title);
+
+        if (description != null) {
+            embedBuilder.setDescription(description);
+        }
+
+        embedBuilder.setColor(new Color(24, 221, 136));
+
+        return embedBuilder.build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        // Get the roles the bot can interact with
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+        List<String> roleIds = guildRoles.stream().map(Role::getId).collect(Collectors.toList());
+
+        List<Role> selectedRoles = new ArrayList<>();
+        for (SelectOption role : Objects.requireNonNull(event.getSelectedOptions())) {
+            if (roleIds.contains(role.getValue())) {
+                selectedRoles.add(guildRoles.get(roleIds.indexOf(role.getValue())));
+            }
+        }
+
+        // True if the event option was 'choose'
+        if (event.getMessage().isEphemeral()) {
+
+            SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+            menu.setPlaceholder("Select your roles").setMaxValues(selectedRoles.size());
+
+            for (SelectOption roleOption : Objects.requireNonNull(event.getSelectedOptions())) {
+                Role role = guildRoles.get(roleIds.indexOf(roleOption.getValue()));
+                menu.addOption(role.getName(), role.getId());
+            }
+
+            event.getChannel()
+                .sendMessageEmbeds(event.getMessage().getEmbeds().get(0))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+
+            return;
+        }
+
+        List<SelectOption> menuOptions =
+                Objects.requireNonNull(event.getInteraction().getComponent()).getOptions();
+
+        // Remove deselected roles
+        for (SelectOption option : menuOptions) {
+            Role role = guildRoles.get(roleIds.indexOf(option.getValue()));
+            if (!selectedRoles.contains(role)) {
+                Objects.requireNonNull(event.getGuild()).removeRoleFromMember(member, role).queue();
+            }
+        }
+
+        // Add the selected roles to the member
+        for (Role role : guildRoles) {
+            if (selectedRoles.contains(role)) {
+                Objects.requireNonNull(event.getGuild()).addRoleToMember(member, role).queue();
```

Comment:
You already run Objects#requireNonNull earlier on the Guild.

Instead of adding "random" requireNonNull in the code, you should validate whatever you can at the start of the method in the first few lines.